### ISSUE TITLE
Hook for mimetype

### ISSUE THREAD

**Issue by giorgiozoppi** [waliarubal/Jaya#68]
I would expect when i click on a file, be asked to open with a mime file or if the plugin is remote (dropbox/drive/s3) to download, i.e.
In the filesystem:
- when i click on image i'll open a default image viewer
- when i click on a video i'll open the default video player.
In a remote whenever i will click, it will show a popup asking me for download.

**Comment by waliarubal** [comment]
Yes, this is exactly the behavior which is planned. I will definitely plan API changes for allowing the plugin developers to implement hooking up the same.
